wrapper/src/test/java/integration/container/tests/TopologyQueryTests.java

```diff
@@ -0,0 +1,275 @@
+      );
+    } else {
+      AuroraMysqlDialect dialect = new AuroraMysqlDialect();
+      query = dialect.getTopologyQuery();
+      expectedTypes = Arrays.asList(
+          "VARCHAR",
+          "BIGINT",
+          "DOUBLE",
+          "DOUBLE",
+          "DATETIME"
+      );
+    }
+
+    final Connection conn = DriverManager.getConnection(url, props);
+    assertTrue(conn.isValid(5));
+    Statement stmt = conn.createStatement();
+    stmt.executeQuery(query);
+    ResultSet rs = stmt.getResultSet();
+    int cols = rs.getMetaData().getColumnCount();
+    List<String> columnTypes = new ArrayList<>();
+    for (int i = 1; i <= cols; i++) {
+      columnTypes.add(rs.getMetaData().getColumnTypeName(i));
+    }
+    assertEquals(expectedTypes, columnTypes);
+    conn.close();
+  }
+
+  @TestTemplate
+  @ExtendWith(TestDriverProvider.class)
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  @EnableOnNumOfInstances(min = 2)
+  public void testAuroraTimestamp(TestDriver testDriver) throws SQLException, ParseException {
+    LOGGER.info(testDriver.toString());
+
+    final Properties props = ConnectionStringHelper.getDefaultPropertiesWithNoPlugins();
+    DriverHelper.setConnectTimeout(testDriver, props, 10, TimeUnit.SECONDS);
+    DriverHelper.setSocketTimeout(testDriver, props, 10, TimeUnit.SECONDS);
+
+    // Get second instance since first one has null timestamps
+    String url =
+        ConnectionStringHelper.getWrapperUrl(
+            testDriver,
+            TestEnvironment.getCurrent()
+                .getInfo()
+                .getDatabaseInfo()
+                .getInstances()
+                .get(1)
+                .getHost(),
+            TestEnvironment.getCurrent()
+                .getInfo()
+                .getDatabaseInfo()
+                .getInstances()
+                .get(1)
+                .getPort(),
+            TestEnvironment.getCurrent().getInfo().getDatabaseInfo().getDefaultDbName());
+    LOGGER.finest("Connecting to " + url);
+
+    if (TestEnvironment.getCurrent().getCurrentDriver() == TestDriver.MARIADB
+        && TestEnvironment.getCurrent().getInfo().getRequest().getDatabaseEngine() == DatabaseEngine.MYSQL) {
+      props.setProperty("permitMysqlScheme", "1");
+    }
+
+    SimpleDateFormat format;
+    if (TestEnvironment.getCurrent().getCurrentDriver() == TestDriver.PG) {
+      AuroraPgDialect dialect = new AuroraPgDialect();
+      query = dialect.getTopologyQuery();
+      format = new SimpleDateFormat("yyy-MM-dd HH:mm:ssX");
+      format.setTimeZone(TimeZone.getTimeZone("GMT"));
+    } else {
+      AuroraMysqlDialect dialect = new AuroraMysqlDialect();
+      query = dialect.getTopologyQuery();
+      format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSSSS");
+    }
+
+    final Connection conn = DriverManager.getConnection(url, props);
+    assertTrue(conn.isValid(5));
+    Statement stmt = conn.createStatement();
+    stmt.executeQuery(query);
```

Comment:
Can you use try with resources here? https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
Same comment for the other locations.